nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -2031,6 +2036,87 @@ protected Object getDataModelField(DocumentRef docRef, String schema, String fie
         return null;
     }
 
+    @Override
+    public void makeRecord(DocumentRef docRef) {
+        Document doc = resolveReference(docRef);
+        if (isRecord(docRef)) {
+            // already a record, don't do anything
+            return;
+        }
+        checkPermission(doc, CAN_MAKE_RECORD);
+        DocumentModel docModel = readModel(doc);
+        notifyEvent(DocumentEventTypes.BEFORE_MAKE_RECORD, docModel, null, null, null, true, false);
+        doc.makeRecord();
```

Comment:
Should we do this action in a separate transaction? I'm thinking about listener rollbacking the action to avoid creation of record in system.

Reply:
Could you expand on the risk / threat scenario?

Reply:
If someone contributes a listener which throws an exception it could break the make record action.

Reply:
This is not really an issue